### DISCUSSION TITLE
a bug in putObject the we always  get 12 data and 4 parity

### DISCUSSION THREAD

**Discussion by aviscode** [minio/minio#13008] [Q&A]
when I put an object i debugged the func putObject in /minio/cmd/erasure-object.go putObject
it will allwas go in to the 
   `if !opts.MaxParity {...}`
(because we call the putObject func from PutObjectHandler in cmd/object-handlers.go
with a new `opts ObjectOptions` that the  **MaxParity  is false** )
and because of it the parityDrives is becoming to =to 4 and we always  get 12 data and 4 parity and if I check the parity before the if its =8 like it should be and that meins that we always get 12data and 4 parity
as u can see in my debug output  

```
parityDrives
8
after globalStorageClass func
0
4
after if
4
12
fi.Erasure.DataBlocks
12
fi.Erasure.ParityBlocks
4
fi.Erasure.BlockSize
```

## Expected Behavior

8 data and 8 parity

## Current Behavior
12 data and 4 parity

## Steps to Reproduce (for bugs)

1. add some printings in the func putObject in /minio/cmd/erasure-object.go putObject.
2.and see the parityDrives

## Your Environment

* Version used (`minio --version`):minio version DEVELOPMENT.GOGET
* Server setup and configuration:./minio server /mnt/disk{1...16}
* Operating System and version (`uname -a`):x86_64 GNU/Linux

**Comment by vadmeste** [comment]
> ## Current Behavior
> 12 data and 4 parity

This is actually expected. Parity is 4 by default. We don't think you need more than 4 parity disks for a given erasure set.

Closing this, please feel free to reopen if needed.

**Comment by aviscode** [comment]
but i set it to be 8
like this 
export MINIO_STORAGE_CLASS_STANDARD=EC:8
export MINIO_STORAGE_CLASS_RRS=EC:8
@vadmeste

**Comment by vadmeste** [comment]
> but i set it to be 8
> like this
> export MINIO_STORAGE_CLASS_STANDARD=EC:8
> export MINIO_STORAGE_CLASS_RRS=EC:8
> @vadmeste

Oh okay, I need to check this in our side.

**Comment by vadmeste** [comment]
@aviscode it works fine for me. 

You can run this to check:
```
go run docs/bucket/versioning/xl-meta.go /tmp/xl/1/testbucket/Dockerfile/xl.meta
```

It shows:
```
        "EcM": 8,
        "EcN": 8,
```

which means parity and data are both 8 disks.

**Comment by aviscode** [comment]
@evandandrea 
go run docs/bucket/versioning/xl-meta.go /mnt/disk1/avi/2.2/xl.meta 
`[root@rack-a-node-2 minio]# go run docs/bucket/versioning/xl-meta.go /mnt/disk1/avi/2.2/xl.meta 
{
  "Versions": [
    {
      "Type": 1,
      "V2Obj": {
        "CSumAlgo": 1,
        "DDir": "CkhUkxaUR+iSd+pP3KUiwA==",
        "EcAlgo": 1,
        "EcBSize": 1048576,
        "EcDist": [
          7,
          8,
          9,
          10,
          11,
          12,
          13,
          14,
          15,
          16,
          1,
          2,
          3,
          4,
          5,
          6
        ],
        "EcIndex": 7,
        "EcM": 12,
        "EcN": 4,
        "ID": "AAAAAAAAAAAAAAAAAAAAAA==",
        "MTime": 1629378329621570299,
        "MetaSys": {
          "x-minio-internal-inline-data": "dHJ1ZQ=="
        },
        "MetaUsr": {
          "content-type": "application/octet-stream",
          "etag": "09896041cccac8f7b0aca6bcce3c3078"
        },
        "PartASizes": [
          37
        ],
        "PartETags": [
          ""
        ],
        "PartNums": [
          1
        ],
        "PartSizes": [
          37
        ],
        "Size": 37
      }
    }
  ]
}`

`   "EcM": 12,
        "EcN": 4,`

and i run before 
export MINIO_STORAGE_CLASS_STANDARD=EC:8
export MINIO_STORAGE_CLASS_RRS=EC:8

**Comment by vadmeste** [comment]
@aviscode this won't apply with existing object, could you upload a new object and check for EcM and EcN again ?

**Comment by harshavardhana** [comment]
You need to set the storage class before starting MinIO , please share your entire command line. 

It looks like you are setting environment variables not where the server is running.

**Comment by aviscode** [comment]
i did

**Reply by harshavardhana** [reply]
> i did

How ?

**Reply by aviscode** [reply]
ok i see u need to run the
export MINIO_STORAGE_CLASS_STANDARD=EC:8
export MINIO_STORAGE_CLASS_RRS=EC:8
in the same shell window 
working great now 
but way if i change it from mc it's not working well?
